File: 1497 Check If Array Pairs Are Divisible by k/1497.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        c = Counter([(x % k + k) % k for x in arr])

        if c[0] % 2 == 1:
            return False

        for i in range(1, k//2 + 1):
            if c[i] != c[k-i]:
                return False

        return True

File: 1497 Check If Array Pairs Are Divisible by k/1497.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {
        vector<int> v(k, 0);
        
        for (int x: arr){
            v[(x % k + k) % k]++;
        }

        if (v[0] % 2 == 1){
            return false;
        }

        for (int i = 1; i <= k/2; i++){
            if (v[i] != v[k-i]){
                return false;
            }
        }

        return true;
    }
};
